packages/playwright-core/src/server/injected/reactSelectorEngine.ts

```diff
@@ -165,12 +172,17 @@ export const ReactEngine: SelectorEngine = {
     const reactRoots = findReactRoots(document);
     const trees = reactRoots.map(reactRoot => buildComponentsTree(reactRoot));
     const treeNodes = trees.map(tree => filterComponentsTree(tree, treeNode => {
+      const props = treeNode.props ?? {};
+
+      if (treeNode.key !== undefined)
+        props.key = treeNode.key;
```

Comment:
can I have a react prop named "key" that would clash with this?

Reply:
you can't :)

Reply:
oh nice! Would you mind also adding a line to the docs on the react selectors regarding `key` filtering?

Reply:
Done @aslushnikov!